### PR TITLE
Add specs for IO#autoclose? and IO#autoclose=

### PR DESCRIPTION
**Pull request by herwinw**
These are the very basic specs, set a value, get a value. The actual behaviour change is not tested, I tried the example of https://ruby-doc.org/3.2.2/IO.html#method-i-autoclose-3D and a couple of variations, and I could not get them to work.

### CODE REVIEW

File: core/io/autoclose_spec.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  it "cannot be queried on a closed IO object" do
+    @io.close
+    -> { @io.autoclose? }.should raise_error(IOError, /closed stream/)
+  end
+
+  it "cannot be set on a closed IO object" do
+    @io.close
+    -> { @io.autoclose = false }.should raise_error(IOError, /closed stream/)
+  end
+end
```
Comment (andrykonchin) [inline core/io/autoclose_spec.rb:49]: Looks good, but it makes sense to have separate test cases for the methods `#autoclose=` and `#autoclose?`.

There could also be added the following checks:
- `nil` value is treated as `false` in `#autoclose=`
- by default `#autoclose?` returns `true` (it's actually describe behaviour of a constructor method)

File: core/io/autoclose_spec.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  end
+
+  it "can be set to false" do
+    @io.autoclose = false
+    @io.should_not.autoclose?
+  end
+
+  it "can be set to any truthy value" do
+    @io.autoclose = 42
+    @io.should.autoclose?
```
Comment (andrykonchin) [inline core/io/autoclose_spec.rb:26]: **suggestion:** It makes sense to set `false` and then to assign truthy value - this way we ensure assigning truthy value isn't ignored and affects an io object.

Also it makes sense to assign different truthy values (`Object.new`, `[]`, `"abc"`, etc) in the same test case as an illustration that there is no special logic for `Integer` type (and Integer value `42`)

**Review by andrykonchin — APPROVED** [review]

**Comment by herwinw** [comment]
(Converted this one to draft, so it's easier for me to see that I have a few review comments I need to apply)

File: core/io/autoclose_spec.rb (hunk 2 above)
Comment (herwinw) [reply core/io/autoclose_spec.rb:33]: I added `Object.new` as well. That, combined with the test description of "any truthy value" should be clear enough.

**Comment by seven1m** [comment]
Was there any additional feedback on this? I wonder if we can merge?

**Comment by andrykonchin** [comment]
This wasn't addressed
> Looks good, but it makes sense to have separate test cases for the methods #autoclose= and #autoclose?.

neither with with changes nor with starting a discussion.

I don't insist on the change but I believe it should be a conscious decision that we are intentionally mixing specs of two methods (`#autoclose` and `#autoclose?`) in the same file and in the same `describe` section.

**Pull request merged**